In [ ]:
import pandas as pd

df = pd.read_csv("../data/Smartphone_Usage_Productivity_Dataset_50000.csv")

df.head()

,User_ID,Age,Gender,Occupation,Device_Type,Daily_Phone_Hours,Social_Media_Hours,Work_Productivity_Score,Sleep_Hours,Stress_Level,App_Usage_Count,Caffeine_Intake_Cups,Weekend_Screen_Time_Hours
0,U1,58,Male,Professional,Android,1.3,6.7,6,8.8,4,42,1,8.7
1,U2,25,Male,Professional,Android,1.2,1.5,5,6.4,1,51,3,5.1
2,U3,19,Male,Student,iOS,5.3,5.7,5,9.0,4,14,5,6.3
3,U4,35,Female,Business Owner,iOS,5.8,2.5,2,5.7,3,36,6,12.8
4,U5,33,Male,Freelancer,Android,7.9,1.3,4,5.7,3,37,5,9.9


In [7]:
df.corr(numeric_only=True)["Work_Productivity_Score"].sort_values(ascending=False)

Work_Productivity_Score      1.000000
Stress_Level                 0.009786
Social_Media_Hours           0.003625
Sleep_Hours                  0.003243
Caffeine_Intake_Cups         0.001253
App_Usage_Count              0.001125
Daily_Phone_Hours           -0.002404
Age                         -0.003480
Weekend_Screen_Time_Hours   -0.005343
Name: Work_Productivity_Score, dtype: float64

Die Korrelation zeigt, dass kein Merkmal einen starken Einfluss auf den Work_Productivity_Score hat, da alle Werte nahe bei 0 liegen.

Dennoch wurden folgende Felder ausgewählt, da sie inhaltlich sinnvoll sind:

- Stress_Level (höchste Korrelation)
- Social_Media_Hours (Ablenkung)
- Sleep_Hours (Erholung)
- Daily_Phone_Hours (Hauptthema der Analyse)

In [8]:
!pip install scikit-learn


[notice] A new release of pip is available: 25.3 -> 26.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [9]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Features & Ziel
X = df[["Stress_Level", "Social_Media_Hours", "Sleep_Hours", "Daily_Phone_Hours"]]
y = df["Work_Productivity_Score"]

# Split
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

# Modell
model = LinearRegression()
model.fit(X_train, y_train)

# Vorhersage
y_pred = model.predict(X_test)

# Fehler
mse = mean_squared_error(y_test, y_pred)

print("MSE:", mse)

MSE: 8.306117898931413


In [10]:
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix, precision_score, recall_score, accuracy_score

# Zielvariable in Klassen umwandeln
y_class = (df["Work_Productivity_Score"] >= 5).astype(int)

# Features bleiben gleich
X = df[["Stress_Level", "Social_Media_Hours", "Sleep_Hours", "Daily_Phone_Hours"]]

# Split
X_train, X_test, y_train, y_test = train_test_split(
    X, y_class, test_size=0.2, random_state=42
)

# Modell
model = LogisticRegression()
model.fit(X_train, y_train)

# Vorhersage
y_pred = model.predict(X_test)

# Auswertung
cm = confusion_matrix(y_test, y_pred)
precision = precision_score(y_test, y_pred)
recall = recall_score(y_test, y_pred)
accuracy = accuracy_score(y_test, y_pred)

print("Confusion Matrix:")
print(cm)
print("Precision:", precision)
print("Recall:", recall)
print("Accuracy:", accuracy)

Confusion Matrix:
[[   0 3986]
 [   0 6014]]
Precision: 0.6014
Recall: 1.0
Accuracy: 0.6014


4.4


Mein Modell funktioniert insgesamt nicht gut.  
Es Genauigkeit von circa 60%, was jetzt nicht besonders gut ist, vorallem weil es einfach immer eine hohe Produktivität vorraussagt.

Der Recall ist mit 1.0 sehr hoch, was bedeutet, dass alle positiven Fälle erkannt werden.  
Die Precision ist jedoch nur etwa 0.60, was zeigt, dass viele Vorhersagen falsch sind.

Also zusammengefasst kann man sagen, dass Modell würde sich nicht für einen täglichen Gebrauch eignen.